File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/main/java/io/github/lishangbu/avalon/dataset/repository/PokemonTypeRepository.java
package io.github.lishangbu.avalon.dataset.repository;

import io.github.lishangbu.avalon.dataset.entity.PokemonType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

/// 宝可梦属性(PokemonType)数据访问层
///
/// 提供基础的 CRUD 操作
///
/// @author lishangbu
/// @since 2025/09/14
@Repository
public interface PokemonTypeRepository
    extends JpaRepository<PokemonType, PokemonType.PokemonTypeId>,
        JpaSpecificationExecutor<PokemonType> {}

File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/main/kotlin/io/github/lishangbu/avalon/dataset/repository/PokemonTypeRepository.kt
package io.github.lishangbu.avalon.dataset.repository

import io.github.lishangbu.avalon.dataset.entity.PokemonType
import io.github.lishangbu.avalon.dataset.entity.PokemonTypeId
import org.babyfish.jimmer.spring.repository.KRepository
import org.springframework.stereotype.Repository

/**
 * 宝可梦属性仓储接口
 *
 * 定义宝可梦属性数据的查询与持久化操作
 *
 * @author lishangbu
 * @since 2025/09/14
 */
@Repository
interface PokemonTypeRepository : KRepository<PokemonType, PokemonTypeId>
